util/stmtsummary/v2/reader.go

```diff
@@ -388,8 +388,8 @@ func (r *HistoryReader) scheduleTasks(
 	case <-ctx.Done():
 		// notified by manager or parent ctx is canceled
 	}
-	close(rowsCh) // task done
 	mgrWg.Wait()
+	close(rowsCh) // task done
```

Comment:
Is the corrupted line caused by this?

Reply:
No. But if err occurs, closing `rowsCh` before workers are done can cause sending to a closed channel and then panic.